tests/unit/query_v2_create_expand_multiframe.cpp

```diff
@@ -0,0 +1,89 @@
+
+#include "query/v2/bindings/frame.hpp"
+#include "query/v2/bindings/symbol_table.hpp"
+#include "query/v2/common.hpp"
+#include "query/v2/context.hpp"
+#include "query/v2/plan/operator.hpp"
+#include "query/v2/requests.hpp"
+#include "storage/v3/property_value.hpp"
+#include "storage/v3/shard.hpp"
+#include "utils/logging.hpp"
+#include "utils/memory.hpp"
+
+namespace memgraph::query::v2 {
+
+MultiFrame CreateMultiFrame(const size_t max_pos, const Symbol &src, const Symbol &dst, MockedRequestRouter *router) {
+  static constexpr size_t frame_size = 100;
+  MultiFrame multi_frame(max_pos, frame_size, utils::NewDeleteResource());
+  auto frames_populator = multi_frame.GetInvalidFramesPopulator();
+  size_t i = 0;
+  for (auto &frame : frames_populator) {
+    frame.MakeValid();
+    auto &src_acc = frame.at(src);
+    auto &dst_acc = frame.at(dst);
+    auto v1 = msgs::Vertex{.id = {{msgs::LabelId::FromUint(1)}, {msgs::Value(static_cast<int64_t>(i++))}}};
+    auto v2 = msgs::Vertex{.id = {{msgs::LabelId::FromUint(1)}, {msgs::Value(static_cast<int64_t>(i++))}}};
+    std::map<msgs::PropertyId, msgs::Value> mp;
+    src_acc = TypedValue(query::v2::accessors::VertexAccessor(v1, mp, router));
+    dst_acc = TypedValue(query::v2::accessors::VertexAccessor(v2, mp, router));
+  }
+
+  return multi_frame;
+}
+
+TEST(CreateExpandTest, Cursor) {
+  using testing::_;
+  using testing::Return;
+
+  AstStorage ast;
+  SymbolTable symbol_table;
+
+  plan::NodeCreationInfo node;
+  plan::EdgeCreationInfo edge;
+  edge.edge_type = msgs::EdgeTypeId::FromUint(1);
+  edge.direction = EdgeAtom::Direction::IN;
+  edge.symbol = symbol_table.CreateSymbol("e", true);
+  auto id_alloc = IdAllocator(0, 100);
+
+  const auto &src = symbol_table.CreateSymbol("n", true);
+  node.symbol = symbol_table.CreateSymbol("u", true);
+
+  auto once_cur = plan::MakeUniqueCursorPtr<MockedCursor>(utils::NewDeleteResource());
+  EXPECT_CALL(BaseToMock(*once_cur), PullMultiple(_, _)).Times(1);
+
+  std::shared_ptr<plan::LogicalOperator> once_op = std::make_shared<MockedLogicalOperator>();
+  EXPECT_CALL(BaseToMock(*once_op), MakeCursor(_)).Times(1).WillOnce(Return(std::move(once_cur)));
+
+  auto create_expand = plan::CreateExpand(node, edge, once_op, src, true);
+  auto cursor = create_expand.MakeCursor(utils::NewDeleteResource());
+
+  MockedRequestRouter router;
+  EXPECT_CALL(router, CreateExpand(_))
+      .Times(1)
+      .WillOnce(Return(std::vector<msgs::CreateExpandResponse>{msgs::CreateExpandResponse{}}));
+  auto context = MakeContext(ast, symbol_table, &router, &id_alloc);
+  auto multi_frame = CreateMultiFrame(context.symbol_table.max_position(), src, node.symbol, &router);
+  cursor->PullMultiple(multi_frame, context);
+
+  auto frames = multi_frame.GetValidFramesReader();
+  for (auto &frame : frames) {
+    EXPECT_EQ(frame[edge.symbol].IsEdge(), true);
+    const auto &e = frame[edge.symbol].ValueEdge();
+    EXPECT_EQ(e.EdgeType(), edge.edge_type);
+  }
```

Comment:
Two things here:
1. The number of valid frames are not checked.
2. maybe it worth to populate some part of the multiframe, not the whole to make sure the cursor works well in that case also. Of course, the invalid frames than also have to be verified.

Reply:
This should be fixed now